test/unit/dom/dom.js

```diff
@@ -1,68 +1,110 @@
 /* global testSketchWithPromise */
 suite('DOM', function() {
   suite('p5.prototype.select', function() {
-    var myp5;
+    /**
+     * Uses p5 in instance-mode inside a custom container.
+     * All elements are attached inside the container for testing
+     * And cleaned up on teardown.
+     */
+    let myp5;
+    let myp5Container;
 
     setup(function(done) {
+      myp5Container = document.createElement('div');
+      document.body.appendChild(myp5Container);
       new p5(function(p) {
         p.setup = function() {
           myp5 = p;
           done();
         };
-      });
+      }, myp5Container);
     });
 
     teardown(function() {
       myp5.remove();
+      if (myp5Container && myp5Container.parentNode) {
+        myp5Container.parentNode.removeChild(myp5Container);
+      }
+      p5Container = null;
     });
 
-    var elt;
+    test('should return only one p5.element if match is found', function() {
+      // Adding 2 buttons to container
+      myp5.createCheckbox('Text 1');
+      myp5.createCheckbox('Text 2');
+      const result = myp5.select('input');
 
-    teardown(function() {
-      if (elt && elt.parentNode) {
-        elt.parentNode.removeChild(elt);
-        elt = null;
-      }
+      assert.instanceOf(result, p5.Element);
     });
 
-    test('should find elements by ID', function() {
-      elt = document.createElement('div');
-      elt.setAttribute('id', 'blarg');
-      document.body.appendChild(elt);
+    const generateButton = (name, className = null) => {
+      const button = myp5.createButton(name);
+      if (className) {
+        button.class(className);
+      }
+      return button;
+    };
 
-      assert.strictEqual(myp5.select('#blarg').elt, elt);
-    });
+    test('should find element by class name', function() {
+      // Creates 2 buttons with same class and test if it selects only one.
+      const testClassName = 'test-button';
+      const testButton1 = generateButton('Button 1', testClassName);
+      // eslint-disable-next-line no-unused-vars
+      const testButton2 = generateButton('Button 2', testClassName);
 
-    test('should return null when elements by ID are not found', function() {
-      assert.isNull(myp5.select('#blarg'));
+      const result = myp5.select(`.${testClassName}`);
+      assert.deepEqual(result.elt, testButton1.elt);
     });
 
-    test('should find elements by class', function() {
-      elt = document.createElement('div');
-      elt.setAttribute('class', 'blarg');
-      document.body.appendChild(elt);
+    test('should find element by class name from given container', function() {
+      // Creates 2 buttons with same class in a container and test if it selects only one.
+      const testClassName = 'test-button';
+      // eslint-disable-next-line no-unused-vars
```

Comment:
I don't get the need for this hack here :sweat_smile: . If you are not using variable `testButton1` later on and knowing the scope of `const`, wouldn't `generateButton('Button 1', testClassName);` suffice instead of `const testButton1 = generateButton('Button 1', testClassName);`. Same thing for everywhere `eslint-disable` is used in this file?

Reply:
Yup, it would. didn't notice that.